src/main/notifications/index.ts

```diff
@@ -97,10 +78,33 @@ class NotificationManager {
             this.allActiveNotifications.delete(mention.uId);
         });
 
-        mention.on('failed', () => {
-            this.allActiveNotifications.delete(mention.uId);
+        return new Promise((resolve) => {
```

Comment:
I wonder if there is any chance this promise never gets resolved. I mean... are we positive by calling `mention.show` either the show or the failed code is going to execute? Is there any chance they won't (or take a worrying amount of time)?

If that were the case, we could add some timeout to resolve this if it takes longer than X (30s?).

Reply:
I was thinking that, and that would indicate a clear failure, but wouldn't really add more information in terms of metrics. If we never get an ACK as a result of this Promise not returning, then we can count that as a missed notification since we can't guarantee that `show()` was called or finished correctly. I think the `failed()` callback handles most of that, but the lack of an ACK should be just as good of an indication that something went wrong, and the metrics will reflect that as part of notification health.

Reply:
I am not so much worried about the metrics, but the memory leak of promises being created and never getting resolved.

Reply:
Ahh, that's a good point. In that case I think a timeout is warranted.

Reply:
This is now a "purist nit", so feel free to ignore.

Resolve is supposed to be used when the promise ends correctly. In theory, when it fails, we should reject the promise.

Nevertheless, I was trying to look in the internet some good reason to do it one way or another, and I think in this case would only add unneeded boilerplate to it :P

Reply:
Yeah also `reject` commonly causes an error state which is sort of the case, but not what we're trying to achieve here.